babel_monitor/src/lib.rs

```diff
@@ -196,31 +196,96 @@ pub fn parse_interfaces(stream: &mut TcpStream) -> Result<Vec<Interface>, BabelM
     parse_interfaces_sync(babel_output)
 }
 
+/// Gets this routers local fee, what the router charges for bandwidth. The unit is wei (1*10-18 of a dollar) per byte
 pub fn get_local_fee(stream: &mut TcpStream) -> Result<u32, BabelMonitorError> {
     let output = run_command(stream, "dump")?;
 
     let babel_output = output;
     get_local_fee_sync(babel_output)
 }
 
+/// Sets this routers local fee, what the router charges for bandwidth. The unit is wei (1*10-18 of a dollar) per byte
 pub fn set_local_fee(stream: &mut TcpStream, new_fee: u32) -> Result<(), BabelMonitorError> {
     let result = run_command(stream, &format!("fee {new_fee}"))?;
 
     let _out = result;
     Ok(())
 }
 
+/// Sets the metric factor for babel. This is a weighting value used to decide if this router should select
+/// routes based on price or quality of service. A higher value will cause the router to prefer routes with
+/// higher quailty of service, a lower value will cause the router to prefer routes with lower price.
 pub fn set_metric_factor(stream: &mut TcpStream, new_factor: u32) -> Result<(), BabelMonitorError> {
     let result = run_command(stream, &format!("metric-factor {new_factor}"))?;
 
     let _out = result;
     Ok(())
 }
 
-pub fn monitor(stream: &mut TcpStream, iface: &str) -> Result<(), BabelMonitorError> {
-    let command = &format!("interface {iface} max-rtt-penalty 2000 enable-timestamps true");
-    let iface = iface.to_string();
-    let result = run_command(stream, command)?;
+/// Sets the interval at which Babel will update it's routes from the kernel routing table. If set to zero Babel will only recieve
+/// updates from the kernel as changes are made and will never perform a full dump.
+pub fn set_kernel_check_interval(
+    stream: &mut TcpStream,
+    kernel_check_interval: Option<Duration>,
+) -> Result<(), BabelMonitorError> {
+    let interval = match kernel_check_interval {
+        // unit is centiseconds
+        Some(d) => (d.as_millis() / 100) as u16,
+        None => 0,
+    };
+    let result = run_command(stream, &format!("kernel-check-interval {interval}"))?;
+
+    let _out = result;
+    Ok(())
+}
+
+/// Sets the default interface parameters for babel. These are applied at startup and can be overridden per interface, note if modified
+/// at runtime then existing interfaces will not be updated.
+pub fn set_interface_defaults(
+    stream: &mut TcpStream,
+    defaults: BabeldInterfaceConfig,
+) -> Result<(), BabelMonitorError> {
+    let mut command = "default ".to_string();
+    command.push_str(&build_interface_config_string(defaults));
+    let result = run_command(stream, &command)?;
+
+    let _out = result;
+    Ok(())
```

Comment:
Do we not want to log these results (L239 as well)

Reply:
probably a good call will do.

Reply:
oh actually the response in this case is 'ok' or 'bad', which us handling is probably a good idea...